pkg/build/builder/common.go

```diff
@@ -66,6 +66,25 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildStatus(c client.BuildInterface, build *api.Build) {
```

Comment:
updateBuildFailureReason seems like a better name for what this does.

but rather than have it actually make the update call, perhaps it would be better if it just updated the build object, and the caller should then perform the update call?  (same thing for updateBuildRevision which you've copied here).